### ISSUE TITLE
Confusing UI

### ISSUE THREAD

**Issue by jscham** [lupoDharkael/flameshot#573]
I find the flameshot UI very confusing and dysfunctional:
- All icons are round and violet. I cannot just select what I want, I have to stop and understand them first.
- They are moving around as the selection changes, so I can't even memorize their positions.
- Especially the X icon is placed among the others like any others and a casual user may wonder if it is a tool and use it by mistake.
- I definetely prefer text to icons anyway (similar to Greenshot). Icons are not so distinctive (check "marker" vs "pencil" or the "open app" too similar to "upload to Imgur"). At first, I thought the four radial arrows icon was supposed to be dragged around to just reposition the selection (I believe I saw this behaviour in some other application) and the selection size icon to set an exact size manually. These icons are more or less passive and they are the only ones. They are odd. "Undo" turns counterclockwise while "Redo" turns clockwise and I don't see any special reason for that (apart from the clock metaphore, which is a bit sophisticated if you are in a hurry). Tooltips are not enough.
- "Undo" is right there even when there is nothing to undo, the same for "Redo".
- Instead of instructions to use a color picker, I would prefer an icon/voice to just select the color. Plus, the right click instructions are not working for me because easystroke interferes with it.

**Comment by ZetaoYang** [comment]
I aggree with the second one.

**Comment by nicekiwi** [comment]
I thought the same when I first use it, but once I'd take a few screen shots it was easy enough to know at a glance which icon does what and easily access the one i wanted.

**Comment by sparcbr** [comment]
I find the UI very confusing as well. 

I mostly use copy to clipboard, save and a few others. Maybe there could be a way to group the 
buttons and/or highlight the most used.

Also, there could be keyboard shortcuts. Haven't found anything about this on the configuration.

**Comment by nicekiwi** [comment]
There are keyboard shortcuts and you can choose which icons you see in the preferences. Check the documentation and the -help command.

**Comment by Murmele** [comment]
> I mostly use copy to clipboard, save and a few others. Maybe there could be a way to group the
> buttons and/or highlight the most used.

Or maybe just hiding the not needed icons? Changing the order of the tools, because the pencil is always at the same position?

But it is an amazing application. Thank you!

**Comment by Tset-Noitamotua** [comment]
The moving icons is a great idea! This prevents issues with tools not being accessible when you make screenshots near the edge of your display. Have experienced this often with other screenshot tools and always just thought - WTF!

I found the UX with this app very intuitive from the very first try.

Awesome application :heart:

**Comment by fikrillah** [comment]
maybe we can make the icons on one line only, so no matter if we take small or big screenshot, we can remember the position of the icons...

**Comment by fikrillah** [comment]
also, selected tool is not distinctive enough, i mean it's hard to know which tool that is active at a time

**Comment by fikrillah** [comment]
ok so, i made the ui concept as attempt to solve this issue.

(sorry for badly drawn picture)
![flameshot feature  mockup2](https://user-images.githubusercontent.com/48489341/65430866-d12b6280-de42-11e9-907b-5da2a14f20a4.png)

that's for drawing tool, and here's contextual menu for other  tool:

LINE 
- two button for selecting normal and highlight tool
- if normal line selected, there should be option for arrow
- round/boxy end option
- size and color like in the picture

SQUARE
- option for stroke, fill, and blur 
- if stroke selected, there should be thickness control
- if blur selected, intensity control
- roundness level control
- if stroke and fill selected, color present

CIRCLE
- same as square maybe, but stroke menu should be there

ERASER
- size
- hardness (not necessary actually)

note:
- all component of the tool bar should be less standout than selected screenshot area
- all contextual menu should be easy to discover/understood
- toolbar shrinks when side panel activated

**Comment by D4rkSl4ve** [comment]
I would give up 2 colors for WHITE and BLACK.  White is used many times to quickly erase (not blur) out of the screen, and black, well to cover and make it noticeable.

**Comment by pvonmoradi** [comment]
> I would give up 2 colors for WHITE and BLACK. White is used many times to quickly erase (not blur) out of the screen, and black, well to cover and make it noticeable.

Add this line to your `~/.config/flameshot/flameshot.ini`
```ini
userColors=#800000, #ff0000, #ffff00, #00ff00, #008000, #00ffff, #0000ff, #ff00ff, #800080, #ffffff, #000000
```

**Comment by everyx** [comment]
>
>     * All icons are round and violet. I cannot just select what I want, I have to stop and understand them first.
> 
>     * They are moving around as the selection changes, so I can't even memorize their positions.

I agree too much. In addition, the size of the icon is too large. Using a square icon can put more icons in the same row, similar to another screenshot tool [Snipaste](https://www.snipaste.com/).

![图片](https://user-images.githubusercontent.com/1665522/200981077-0e51dc49-674a-4906-815c-2981e0fb519c.png)

**Comment by xahon** [comment]
Issue on placing buttons in a straight line, check out this PR #3199
I noticed some room for cleaning up the code there but it should work as OP wanted

**Comment by sglbl** [comment]
> maybe we can make the icons on one line only, so no matter if we take small or big screenshot, we can remember the position of the icons...
> 
> this is what i mean ![should like this](https://user-images.githubusercontent.com/48489341/65423562-8f46f000-de33-11e9-9ae2-efeb5ccdcd61.png) (the icons is above the selection if we take lower screen)
> 
> it's better than what we got now ![not like this](https://user-images.githubusercontent.com/48489341/65423628-b9001700-de33-11e9-94c4-55714614cc26.png)

I definetely prefer icons over text, but I agree with this one. Same position would help
